src/index.js

```diff
@@ -4,50 +4,46 @@ const metascraper = require("metascraper")([
  require("metascraper-image")(),
  require("metascraper-logo-favicon")(),
 ]);
-require("dotenv").config?.();
+require("dotenv").config();
 
 const express = require("express");
-var cors = require("cors");
+const cors = require("cors");
+const axios = require("axios");
 const { getScrapingConfig } = require("./proxyConfig");
+const twitter = require("twitter-text");
 
 const app = express();
 const port = process.env.PORT || 8090;
-
-const got = require("got");
-const twitter = require("twitter-text");
-
-require("dotenv").config();
-
 const REQUEST_TIMEOUT_MS =
- parseInt(process.env.REQUEST_TIMEOUT_SEC || 40) * 1000;
+ parseInt(process.env.REQUEST_TIMEOUT_SEC || "40") * 1000;
+const MAX_HTML_SIZE = 1000000;
+
+app.use(cors());
 
 app.listen(port, () => {
- console.log(`Server started on port ${port}`);
+ console.log(`✅ Server started on port ${port}`);
 });
 
-app.use(cors());
+app.get("/fetchUrlMeta", async (req, res) => {
+ try {
+  const { url } = req.query;
+  if (!url) return res.status(400).json({ error: "Missing 'url' parameter" });
 
-app.get("/fetchUrlMeta", (req, res) => {
- const { url } = req.query;
- dispatch({ data: [url], cmd: "fetchMetadata" }).then((response) => {
+  const response = await dispatch({ data: [url], cmd: "fetchMetadata" });
   res.json(response);
- });
+ } catch (error) {
+  console.error("Error processing request:", error);
+  res.status(500).json({ error: "Failed to fetch metadata" });
+ }
 });
 
 async function dispatch({ cmd, data }) {
```

Comment:
why does this dispatch function exist? it looks like you are only using fetchMetadata()